x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/AbstractPhysicalOperationProviders.java

```diff
@@ -257,16 +257,26 @@ private void aggregatesToFactory(
                                 );
                             }
                         } else {
-                            Attribute attr = Expressions.attribute(field);
-                            // cannot determine attribute
-                            if (attr == null) {
-                                throw new EsqlIllegalArgumentException(
-                                    "Cannot work with target field [{}] for agg [{}]",
-                                    field.sourceText(),
-                                    aggregateFunction.sourceText()
-                                );
+                            List<Expression> inputExpressions = new ArrayList<>();
```

Comment:
The rate aggregation requires two inputs: the counter field and the timestamp field.